### PR TITLE
Created Subject-Area Specific Units Rule in Program Template Creation

### PR DESCRIPTION
**Pull request by spencerj411**
Aims to accomplish user story #14 . This PR gets all the course codes using the search api, creates a set of subject areas from the codes and gets displayed in a dropdown menu.
**in the midst of creating the template**
image
**after the template has been made**
image

### CODE REVIEW

**Review by Daniel-Jang — APPROVED** [review]
Seems to work fine for me.

File: cassdegrees/static/js/rules.js [hunk 1]
```diff
@@ -3,6 +3,7 @@
 // Translation table between internal names for components and human readable ones.
 const COMPONENT_NAMES = {
     'subplan': "Subplan",
+    'subject_area_specific': "Subject-Area Specific Units",
```
Comment (JackSekoranja) [inline cassdegrees/static/js/rules.js]:
```suggestion
    'subject_area': "Subject-Area Units",
```

Replace use of 'specific' here, as I think it unnecessarily complicates the terminology (NOTE: must accept all changes or this won't work)

File: cassdegrees/static/js/rules.js [hunk 2]
```diff
@@ -194,6 +195,73 @@ Vue.component('rule_course', {
     template: '#courseRequirementTemplate'
 });
 
+Vue.component('rule_subject_area_specific', {
```
Comment (JackSekoranja) [inline cassdegrees/static/js/rules.js]:
```suggestion
Vue.component('rule_subject_area', {
```

Remove use of 'specific' as described above

File: cassdegrees/static/js/rules.js [hunk 3]
```diff
@@ -194,6 +195,73 @@ Vue.component('rule_course', {
...
+        // https://michaelnthiessen.com/force-re-render/
+        do_redraw: function() {
+            this.redraw = true;
+
+            this.$nextTick(() => {
+                this.redraw = false;
+            });
+        }
+    },
+    template: '#subjectAreaSpecificRuleTemplate'
```
Comment (JackSekoranja) [inline cassdegrees/static/js/rules.js]:
```suggestion
    template: '#subjectAreaRuleTemplate'
```

Remove use of 'specific' as described above

File: cassdegrees/templates/widgets/rules.html [hunk 4]
```diff
@@ -58,6 +58,31 @@
     </fieldset>
 </script>
 
+<script type="text/x-template" id="subjectAreaSpecificRuleTemplate">
```
Comment (JackSekoranja) [inline cassdegrees/templates/widgets/rules.html]:
```suggestion
<script type="text/x-template" id="subjectAreaRuleTemplate">
```

Remove use of 'specific' as described above

**Review by JackSekoranja — CHANGES_REQUESTED** [review]
I'd like it if we didn't use the word 'specific' here, as I personally feel it is a poor choice of words. I've removed it in my following suggestions, and have tested that it works.

**Review by JackSekoranja — APPROVED** [review]
Looks good :)

**Pull request merged**